### PR TITLE
fix: Confirm sign in options

### PR DESCRIPTION
**Pull request by haverchuck**
*Issue #, if available:*
#532 

*Description of changes:*
Adds support for ConfirmSignInOptions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Jordan-Nelson — APPROVED** [review]
lgtm

File: packages/amplify_api/android/build.gradle [hunk 1]
```diff
@@ -55,8 +55,8 @@ dependencies {
     api amplifyCore
 
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation "com.amplifyframework:aws-api:1.17.3"
-    implementation "com.amplifyframework:aws-api-appsync:1.17.3"
+    implementation "com.amplifyframework:aws-api:1.17.4"
+    implementation "com.amplifyframework:aws-api-appsync:1.17.4"
```
Comment (Ashish-Nanda) [inline packages/amplify_api/android/build.gradle:59]: Nit: Inconsistent use of single and double quotes across the `build.gradle` files

File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterConfirmSignInRequest.kt [hunk 2]
```diff
@@ -17,10 +17,20 @@ package com.amazonaws.amplify.amplify_auth_cognito.types
...
 import com.amazonaws.amplify.amplify_core.exception.InvalidRequestException
+import com.amplifyframework.auth.cognito.options.AWSCognitoAuthConfirmSignInOptions
 
 data class FlutterConfirmSignInRequest(val map: HashMap<String, *>) {
   val confirmationCode: String = map["confirmationCode"] as String;
-  val options: HashMap<String, *>? = map["options"] as HashMap<String, *>?;
+  val options: AWSCognitoAuthConfirmSignInOptions = formatOptions(map["options"] as HashMap<String, *>?)
+
+  private fun formatOptions(rawOptions: HashMap<String, *>?): AWSCognitoAuthConfirmSignInOptions {
+    var options =  AWSCognitoAuthConfirmSignInOptions.builder();
```
Comment (Ashish-Nanda) [inline packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterConfirmSignInRequest.kt:27]: Should we change this to `val`?

File: packages/amplify_auth_cognito/ios/Classes/AuthCognitoBridge.swift [hunk 3]
```diff
@@ -86,7 +86,8 @@ class AuthCognitoBridge {
     }
     
     func onConfirmSignIn(flutterResult: @escaping FlutterResult, request: FlutterConfirmSignInRequest) {
-        _ = Amplify.Auth.confirmSignIn(challengeResponse: request.confirmationCode) { response in
+
```
Comment (Ashish-Nanda) [inline packages/amplify_auth_cognito/ios/Classes/AuthCognitoBridge.swift:89]: Nit: Extra line

File: packages/amplify_datastore/android/build.gradle [hunk 4]
```diff
@@ -57,8 +57,8 @@ android {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation "com.amplifyframework:aws-datastore:1.17.3"
-    implementation "com.amplifyframework:aws-api-appsync:1.17.3"
+    implementation "com.amplifyframework:aws-datastore:1.17.4"
+    implementation "com.amplifyframework:aws-api-appsync:1.17.4"
```
Comment (Ashish-Nanda) [inline packages/amplify_datastore/android/build.gradle:61]: I think we don't need this dependency on `aws-api-appsync` now that we have the API category correct? We should probably remove this in a separate PR as its not related to this change.

**Review by Ashish-Nanda — APPROVED** [review]
Looks good. Left some minor comments.

File: packages/amplify_datastore/android/build.gradle (hunk 4 above)
Comment (haverchuck) [reply packages/amplify_datastore/android/build.gradle:61]: Agree - we can take a look at that separately.

File: packages/amplify_api/android/build.gradle (hunk 1 above)
Comment (haverchuck) [reply packages/amplify_api/android/build.gradle:59]: We can fix this another time.

File: packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterConfirmSignInRequest.kt (hunk 2 above)
Comment (haverchuck) [reply packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FlutterConfirmSignInRequest.kt]: Fixed.

File: packages/amplify_auth_cognito/ios/Classes/AuthCognitoBridge.swift (hunk 3 above)
Comment (haverchuck) [reply packages/amplify_auth_cognito/ios/Classes/AuthCognitoBridge.swift]: Fixed.

**Review by Ashish-Nanda — APPROVED** [review]
Approved 🚀

**Pull request merged**